pymemcache/client/hash.py

```diff
@@ -383,7 +382,10 @@ def set_many(self, values, *args, **kwargs):
             client_batches[client.server][key] = value
 
         for server, values in client_batches.items():
-            client = self.clients['%s:%s' % server]
+            if isinstance(server, six.string_types):
+                client = self.clients[server]
+            else:
+                client = self.clients['%s:%s' % server]
```

Comment:
I wonder whether these conditions would be slightly better if they were inverted and used `isinstance(server, tuple)` instead. That's the stronger precondition/invariant given the `%` formatting performed in that branch. What do you think?